rabbitmq/src/urabbitmq/impl/amqp_channel.hpp

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include <memory>
+
+#include <userver/engine/deadline.hpp>
+#include <userver/utils/assert.hpp>
+
+#include <engine/ev/thread_control.hpp>
+
+#include <userver/urabbitmq/typedefs.hpp>
+#include <userver/utils/flags.hpp>
+
+#include <amqpcpp.h>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace urabbitmq {
+class ConsumerBaseImpl;
+
+namespace statistics {
+class ConnectionStatistics;
+}
+}  // namespace urabbitmq
+
+namespace urabbitmq::impl {
+
+class IAmqpChannel {
+ public:
+  virtual ~IAmqpChannel() = default;
+
+  virtual void DeclareExchange(const Exchange&, Exchange::Type,
```

Comment:
It would be nice to move the body of at least once function into cpp file to avoid vtable instantiation in eqch translation unit that includes this file